GG/src/Font.cpp

```diff
@@ -1097,61 +1242,29 @@ void Font::ThrowBadGlyph(const std::string& format_str, boost::uint32_t c)
     throw BadGlyph(boost::io::str(boost::format(format_str) % boost::io::str(format % c)));
 }
 
-Pt Font::DetermineLinesImpl(const std::string& text,
-                            Flags<TextFormat>& format,
-                            X box_width,
-                            std::vector<LineData>& line_data,
-                            std::vector<boost::shared_ptr<TextElement> >* text_elements_ptr) const
-{
-    ValidateFormat(format);
-
-#if DEBUG_DETERMINELINES
-    std::cout << "Font::DetermineLines(text=\"" << text << "\" (@ "
-              << static_cast<const void*>(&*text.begin()) << ") format="
-              << format << " box_width=" << box_width << ")" << std::endl;
-#endif
-
-    std::vector<boost::shared_ptr<TextElement> > local_text_elements;
-
-    std::vector<boost::shared_ptr<TextElement> >& text_elements =
-        text_elements_ptr ? *text_elements_ptr : local_text_elements;
-
+void Font::FillTextElements(const std::string& text,
+                            bool ignore_tags,
+                            std::vector<boost::shared_ptr<TextElement> >& text_elements) const {
     if (text_elements.empty()) {
         using namespace boost::xpressive;
 
-        std::stack<Substring> tag_stack;
-        bool ignore_tags = format & FORMAT_IGNORETAGS;
-        MatchesKnownTag matches_known_tag(KnownTags(), ignore_tags);
-        MatchesTopOfStack matches_tag_stack(tag_stack, ignore_tags);
+        // Fetch and use the regular expression from the TagHandler which parses all the known XML tags.
+        sregex & regex = StaticTagHandler().Regex(text, ignore_tags);
+        sregex_iterator it(text.begin(), text.end(), regex);
 
+        // These are the types found by the regular expression: XML open/close tags, text and
```

Comment:
No need for change but those are not XML tags as they don't follow the specification for XML especially for attributes but 'something else'.